### ISSUE TITLE
Running Sumatra in windows command shell

### ISSUE THREAD

**Issue by apdavison** [fipymigrate/sumatra#111]
Provide windows executables for smt and smtweb commands. Now it is not possible to run Sumatra from standard windows shell.
        

 _Imported from Bitbucket issue [107](https://bitbucket.org/apdavison/sumatra/issue/107),  created by samarkanov on 10-15-2012 at 12:05, last modified: 05-10-2014 at 07:43_

**Comment by apdavison** [comment]
Any ideas how to do this? Use py2exe?
        

_Trac comment by apdavison on 05-15-2013 at 11:26_

**Comment by apdavison** [comment]
I don't know if you are using setuptools, but there you can use the `entry_points`option:

{{{
from setuptools import setup

setup(name=DISTNAME,
      description=DESCRIPTION,
      long_description=LONG_DESCRIPTION,
      maintainer=MAINTAINER,
      maintainer_email=MAINTAINER_EMAIL,
      url=URL,
      license=LICENSE,
      download_url=DOWNLOAD_URL,
      version=VERSION,
      packages=find_packages(exclude=['ez_setup']),
      entry_points={ 
        'console_scripts': [
            'consolecommand = mypackage.mymodule:main_function',
            'anothercmd = mypackage.another_module:cmd_function',
        ]
      },
      )
}}}

This will generate `consolecommand.exe`, `consolecommand-script.py`, `anothercmd.exe`, and `anothercmd-script.py` in your python/Scripts folder.
        

_Trac comment by anonymous on 07-05-2013 at 08:10_

**Comment by apdavison** [comment]
You don't need py2exe or windows .exe files to do this if you want to avoid reliance on setuptools as per the previous comment here. I assume that works and might be the preferred solution. I'll just comment here that Continuum's solution to this is to use Windows .bat batch files, which call the Python scripts. Windows also recognises these as executables. Repeating my comment from the group on how I temporarily fixed this:
https://groups.google.com/forum/?hl=en-GB#!topic/sumatra-users/nXCHJf3Y69Y

"Trying the smt and smtweb scripts installed by sumatra, I find that they do not work in Windows. The setup.py creates smt and smtweb scripts in, in my case, C:\Anaconda\envs\tmm\Scripts
Windows expects executable files to have .bat or .exe extensions, so Continuum's solution to this with Anaconda Python is to create a .bat batch file for each script in the Scripts directory.
In other words, I now create Scripts\smt.bat and Scripts\smtweb.bat alongside the smt and smtweb Python scripts. Both of these contain the identical code:


```
#!text

@echo off
set PYFILE=%~f0
set PYFILE=%PYFILE:~0,-4%
"%~f0\..\..\python.exe" "%PYFILE%" %*
```

These entry-point scripts now allow smt and smtweb to be run from the command line."

If you prefer this to the setup.py entry point solution, you'd still have to get setup.py to add smt.bat and smtweb.bat to the Scripts directory. I don't know why Continuum do it this way, but they have thought very hard about the way they do this, so maybe there's some advantage?

        

_Trac comment by gazzar on 05-10-2014 at 07:43_

**Comment by seankmartin** [comment]
In case it is useful, I used this to get a windows executable with entry points
https://github.com/seankmartin/sumatra/compare/forked-here...seankmartin:master
